validation.py

```diff
@@ -0,0 +1,492 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import glob
+import argparse
+import random
+import functools
+import itertools
+import multiprocessing
+import time
+
+import numpy as np
+import tskit
+# Work around an issue on systems with large numbers of cores.
+# https://github.com/cggh/scikit-allel/issues/285
+os.environ["NUMEXPR_MAX_THREADS"] = f"{os.cpu_count()}"  # NOQA
+import allel
+
+import matplotlib
+matplotlib.use('Agg')  # NOQA # don't try to use $DISPLAY
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+
+import stdpopsim
+import stdpopsim.cli
+
+
+def warning(msg):
+    """
+    Print a warning, with output less ugly than that of warnings.warn().
+    """
+    print(f"WARNING: {msg}", file=sys.stderr)
+
+
+def do_cmd(cmd, out_dir, seed):
+    cmd = cmd.split()
+    assert "-o" not in cmd and "--output" not in cmd
+    assert "-s" not in cmd and "--seed" not in cmd
+    out_file = os.path.join(out_dir, f"{seed}.trees")
+    full_cmd = cmd + f" -q -o {out_file} -s {seed}".split()
+    t0 = time.time()
```

Comment:
This is wall time which isn't the right thing I think. We either want to use ``perf_counter`` or to wrap the command we're calling using ``/usr/bin/time`` to do the work for us. See [here](https://github.com/mcveanlab/treeseq-inference/blob/master/src/evaluation.py#L377) for an example of using time.

Reply:
I'll change it to use perf_counter.

YFI, I got bitten by this (supposedly fixed) GNU time bug last year. https://groups.google.com/forum/#!topic/gnu.utils.help/u1MOsHL4bhg

The fix wasn't included in a release until late 2017 (according to http://ftp.gnu.org/gnu/time/), so there are likely many systems still affected. Unhelpfully, `/usr/bin/time --version` does not report a proper version either, so the only way to check if the bug is present is to allocate a specific amount of memory and see if the /usr/bin/time output is sensible.

Reply:
> YFI, I got bitten by this (supposedly fixed) GNU time bug last year. https://groups.google.com/forum/#!topic/gnu.utils.help/u1MOsHL4bhg
> 
> The fix wasn't included in a release until late 2017 (according to http://ftp.gnu.org/gnu/time/), so there are likely many systems still affected. Unhelpfully, /usr/bin/time --version does not report a proper version either, so the only way to check if the bug is present is to allocate a specific amount of memory and see if the /usr/bin/time output is sensible.

Huh, that is interesting. Looks like a bunch of memory usages I've reported will be off by a factor of 4 then. Oh well.

ps. I looked up [YFI](https://www.internetslang.com/YFI-meaning-definition.asp) to see if this was a new one I hadn't heard of. Interesting slip of the keyboard if it was unintended!

Reply:
RE: YFI; definitely unintended! :)